test/e2e/tests/test_scenarios.py

```diff
@@ -0,0 +1,94 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Tests for custom resources.
+Uses declarative tests framework for custom resources.
+
+To add test: add scenario yaml to scenarios/ directory.
+"""
+
+from declarative_test_fwk import helper, loader, runner
+
+import pytest
+import boto3
+import logging
+
+from e2e import service_bootstrap, service_cleanup, service_marker, scenarios_directory, resource_directory, CRD_VERSION, CRD_GROUP, SERVICE_NAME
+from acktest.k8s import resource as k8s
+
+
+@helper.register_resource_helper(resource_kind="ReplicationGroup", resource_plural="ReplicationGroups")
+class ReplicationGroupHelper(helper.ResourceHelper):
+    """
+    Helper for replication group scenarios.
+    Overrides methods as required for custom resources.
+    """
+
+    def wait_for_delete(self, reference: k8s.CustomResourceReference):
+        logging.debug(f"ReplicationGroupHelper - wait_for_delete()")
+        ec = boto3.client("elasticache")
+        waiter = ec.get_waiter('replication_group_deleted')
+        # throws exception if wait fails
+        waiter.wait(ReplicationGroupId=reference.name)
+
+
+@pytest.fixture(scope="session")
+def input_replacements():
+    """
+    Session scoped fixture to bootstrap service resources and teardown
+    provides input replacements for test scenarios.
+    Eliminates the need for:
+     - bootstrap.yaml and
+     - call to <service_controller_test_dir>/service_bootstrap.py from test-infra
+     - call to <service_controller_test_dir>/service_cleanup.py from test-infra
+    """
+
+    resources_dict = service_bootstrap.service_bootstrap()
```

Comment:
I do not support using `service_bootstrap` and `service_cleanup` inside fixtures. These resources are especially designed to last longer than the scope of a single test invocation. That is, you create them once, run several PyTest runs, and then clean them up. 

I would encourage you to use `get_bootstrap_resources` here, instead: https://github.com/aws-controllers-k8s/elasticache-controller/blob/main/test/e2e/bootstrap_resources.py#L35

Reply:
Updated code and used `get_bootstrap_resources`